Remotion/Data/DomainObjects.UnitTests/Mapping/TypeDefinitionChecker.cs

```diff
@@ -20,67 +20,36 @@
 using NUnit.Framework;
 using Remotion.Data.DomainObjects.Mapping;
 using Remotion.Data.DomainObjects.Persistence.Model;
-using Remotion.Data.DomainObjects.Persistence.NonPersistent.Model;
 using Remotion.Data.DomainObjects.Persistence.Rdbms;
 using Remotion.Data.DomainObjects.Persistence.Rdbms.Model;
+using Remotion.Reflection;
 using Remotion.Utilities;
 
 namespace Remotion.Data.DomainObjects.UnitTests.Mapping
 {
-  public class ClassDefinitionChecker
+  public class TypeDefinitionChecker
   {
-    public void Check (ClassDefinition expectedDefinition, ClassDefinition actualDefinition)
+    public void Check (TypeDefinition expectedDefinition, TypeDefinition actualDefinition)
     {
       ArgumentUtility.CheckNotNull("expectedDefinition", expectedDefinition);
       ArgumentUtility.CheckNotNull("actualDefinition", actualDefinition);
 
-      Assert.AreEqual(expectedDefinition.ID, actualDefinition.ID, "IDs of class definitions do not match.");
-
-      Assert.AreEqual(
-          expectedDefinition.ClassType,
-          actualDefinition.ClassType,
-          "ClassType of class definition '{0}' does not match.",
-          expectedDefinition.ID);
-
-      Assert.AreEqual(
-          expectedDefinition.IsClassTypeResolved,
-          actualDefinition.IsClassTypeResolved,
-          "IsClassTypeResolved of class definition '{0}' does not match.",
-          expectedDefinition.ID);
-
-      Assert.AreEqual(
-          expectedDefinition.IsAbstract,
-          actualDefinition.IsAbstract,
-          "IsAbstract of class definition '{0}' does not match.",
-          expectedDefinition.ID);
-
-      Assert.AreSame(
-          expectedDefinition.InstanceCreator,
-          actualDefinition.InstanceCreator,
-          "Instance creator of class definition '{0}' is not the expected object.",
-          actualDefinition.ID);
-
-      if (expectedDefinition.BaseClass == null)
+      if (expectedDefinition is ClassDefinition expectedClassDefinition)
       {
-        Assert.IsNull(actualDefinition.BaseClass, "actualDefinition.BaseClass of class definition '{0}' is not null.", expectedDefinition.ID);
+        if (actualDefinition is not ClassDefinition actualClassDefinition)
+          throw new InvalidOperationException("The two type definitions must have a matching type.");
```

Comment:
Isn't that an Assert?

Also, sicne you're touching most (all?) of the lines with assert anyway, do you want to switch them over to Assert.That syntax or should we create a ticket for this? They got left behind during the big mirgration  a few years ago.

Reply:
I don't think that changing the assert pattern here is worth the effort. If you really want to do it I would suggest changing this to a collection of errors that are collected and then reported. This way, one would not have to re-run a test to see what all of the differences are. It could also be used for the upcoming mapping test infrastructure to report all failures at once, which instead of only the first failure/first object.

Reply:
The assert-syntax would just be a concsistency thing. but I don't need to touch it now. So, we just stick with changing the invalidopex to assert.